client/builtin_roles.go

```diff
@@ -0,0 +1,12 @@
+package client
+
+const (
+	Admin    string = "Admin"
+	Deployer string = "Deployer"
+	Planner  string = "Planner"
+	Viewer   string = "Viewer"
+)
+
+func IsBuiltinProjectRole(role string) bool {
+	return role == Admin || role == Deployer || role == Planner || role == Viewer
+}
```

Comment:
those are suitable for **Environment** and **Project** default roles.
for **OrganizationRoles** we have **User**, **Admin**.

why there is no `IsBuiltintRole` for the other scopes?
are we missing tests?

Reply:
This is because of the schema used by resourceTeamProjectAssignment:

```

		Schema: map[string]*schema.Schema{
			"team_id": {
				Type:        schema.TypeString,
				Description: "id of the team",
				Required:    true,
				ForceNew:    true,
			},
			"project_id": {
				Type:        schema.TypeString,
				Description: "id of the project",
				Required:    true,
				ForceNew:    true,
			},
			"role": {
				Type:             schema.TypeString,
				Description:      "the assigned built-in role (Admin, Planner, Viewer, Deployer)",
				Optional:         true,
				ValidateDiagFunc: ValidateRole,
				ExactlyOneOf:     []string{"custom_role_id", "role"},
			},
			"custom_role_id": {
				Type:             schema.TypeString,
				Description:      "id of the assigned custom role",
				Optional:         true,
				ExactlyOneOf:     []string{"custom_role_id", "role"},
				ValidateDiagFunc: ValidateNotEmptyString,
			},
		},
	}
```

It's the only one using role and custom_role_id - instead of just role_id.
I thought of deprecating these two fields and adding a role_id instead... but haven't done so yet.

What we can do, is add a proper validation for the other resources (organization and environment). (They're documented, but there's no schema level enforcement).

Reply:
1. Yeah we need a schema validation for Org and Environment too.
2. `role` or `role_id` should be an `uuid` **_or_** one of the builtins (list depends on scope) [then we can deprecate]

each can be in different PR.

Reply:
choosing `role` for both - probably will make less breaking change

Reply:
will add validation.
I'll deprecate the existing two fields.

This will be done in a follow-up PR.

Reply:
sorry I'm a bit confused, in the platform we don't use role_id at all, is it ok the behaviour here will be different?

Reply:
WDTM by not using the id at all?
we do for be able to assign a custom role (uuid) or default role (constant string)